tests/integration-tests/conftest.py

```diff
@@ -855,7 +858,11 @@ def cfn_stacks_factory(request):
     factory = CfnStacksFactory(request.config.getoption("credential"))
     yield factory
     if not request.config.getoption("no_delete"):
-        factory.delete_all_stacks()
+        excluded_stacks = []
+        if request.config.getoption("retain_ad_stack"):
+            excluded_stacks.append(["vpc", "SimpleAD", "MicrosoftAD"])
```

Comment:
Warning: if you append the list, then excluded_stacks will be a list of lists. You have two alternatives:
1. my_array.append(element1) my_array.append(element2)
2. my_array += [element1, element2]
3. my_array.extend([element1, element_2])